packages/presentation/lib/screen/login/login_bloc.dart

```diff
@@ -78,36 +107,81 @@ class _LoginBloc extends BlocImpl<BaseArguments, LoginData>
   }
 
   @override
-  Future<void> auth() async {
+  String? validateLogin() {
+    if (loginValidation == ValidationErrorType.requiredErrorType) {
+      return SM.current.loginFieldRequired;
+    } else if (loginValidation == ValidationErrorType.minLengthErrorType) {
+      return SM.current.loginFieldInvalid;
+    } else {
+      return null;
+    }
+  }
+
+  @override
+  String? validatePassword() {
+    if (passwordValidation == ValidationErrorType.requiredErrorType) {
+      return SM.current.passwordFieldRequired;
+    } else if (passwordValidation == ValidationErrorType.regexErrorType) {
+      return SM.current.passwordFieldInvalid;
+    } else {
+      return null;
+    }
+  }
+
+  @override
+  void onChangedLogin(String changeLogin) {
+    _loginScreenFormKey.currentState?.validate();
+    loginValidation = null;
+    _updateData();
+  }
+
+  @override
+  void onChangedPassword(String changeLogin) {
+    _loginScreenFormKey.currentState?.validate();
+    passwordValidation = null;
+    _updateData();
+  }
+
+  @override
+  Future<void> login() async {
     final login = _loginController.text;
     final password = _passwordController.text;
```

Comment:
you can move it's code to onChangedPassword and set login and password from value onChanged